packages/sdk/src/messageBackend/backend.ts

```diff
@@ -1,188 +1,25 @@
-            "x-hasura-admin-secret": "yaZj76nCg5q"
-          }
-
-
-        const response = await request(this.getUrl(), query, GoldSkyCoreBackend.fillFilter(f), headers);
-        if (response.subgraph_dispatch > 1) throw new Error(`More than one dispatch found for the query`);
-        const dispatch = response.subgraph_dispatch[0];
-        const update = response.subgraph_update[0];
-        const relay = response.subgraph_update[1];
-        const process = response.subgraph_process[0];
-
-
-        let events: Partial<EventResult> = {
-        };
-
-        if (dispatch) {
-            events.dispatch = {
-                tag: "dispatch",
-                leafIndex: dispatch.leaf_index,
-                destinationAndNonce: dispatch.destination_and_nonce,
-                committedRoot: dispatch.committed_root,
-                message: dispatch.message,
-                messageHash: dispatch.message_hash,
-                transactionHash: dispatch.transaction_hash,
-                timestamp: dispatch.timestamp,
-            };
-        }
-
-        if (update) {
-            events.update = {
-                tag: "update",
-                homeDomain: update.home_domain,
-                oldRoot: update.old_root,
-                newRoot: update.new_root,
-                signature: update.signature,
-                transactionHash: update.transaction_hash,
-                timestamp: update.timestamp,
-            };
-        }
-
-        if (relay) {
-            events.relay = {
-                tag: "update",
-                homeDomain: relay.home_domain,
-                oldRoot: relay.old_root,
-                newRoot: relay.new_root,
-                signature: relay.signature,
-                transactionHash: relay.transaction_hash,
-                timestamp: relay.timestamp,
-            };
-        }
-
-        if (process) {
-            events.process = {
-                tag: "process",
-                transactionHash: process.transaction_hash,
-                timestamp: process.timestamp,
-                messageHash: process.message_hash,
-                success: process.success,
-                returnData: process.return_data,
-            };
-        }
-
-        return events;
-    }
+import { Dispatch } from '../messages/types';
+
+export type EventFilter = {
+    committedRoot: string, 
+    messageHash: string,
+    transactionHash: string;
+  };
+
+
+export default abstract class MessageBackend {
+    abstract getDispatch(tx: string): Promise<Dispatch | undefined>;
+    abstract getMessageHash(tx: string): Promise<string | undefined>;
+    abstract dispatchTx(messageHash: string): Promise<string | undefined>;
+    abstract updateTx(messageHash: string): Promise<string | undefined>;
+    abstract relayTx(messageHash: string): Promise<string | undefined>;
+    abstract processTx(messageHash: string): Promise<string | undefined>;
+    abstract confirmAt(messageHash: string): Promise<Date | undefined>;
```

Comment:
Not sure, because if Bridge or Gov will extend it, they would need to extend the Status, which they need anyways. But let me see. Ill check pros and cons I guess.
Good idea though